### PR TITLE
Changes Discord library to Pycord and converts all current commands to slash commands

### PR DESCRIPTION
**Pull request by nathan-jungels**
Added requirements.txt to specify library. Added debug_guild variable. Make sure to run without debug_guild when running in production.

### CODE REVIEW

**Comment by nathan-jungels** [comment]
Now resolves issue #172

File: requirements.txt [hunk 1]
```diff
@@ -0,0 +1,4 @@
+sqlalchemy
+py-cord
+requests
+python-lsp-server
```
Comment (jmercouris) [inline requirements.txt:4]: Is the python-lsp-server here for development?

File: source/utility.py [hunk 2]
```diff
@@ -49,7 +49,8 @@ def str_from_list(urls):
 
 # Create a discord client
 intents = discord.Intents.default()
-client = discord.Client(intents=intents)
+intents.message_content = True
+client = discord.Bot(intents = intents, debug_guilds = [configuration.get('configuration', 'debug_guild')])
```
Comment (jmercouris) [inline source/utility.py:53]: Is there a sample of the `debug_guild` value? What is expected here? What's the configuration.ini look like in this setup?

File: source/bouncer.py [hunk 3]
```diff
@@ -81,77 +86,58 @@ async def process_message(message):
...
+        url_object = urlparse(url)
+        session.query(AllowDomain).filter_by(
+            hostname=url_object.hostname,
+            server_id=ctx.guild.id).delete()
+    session.commit()
+    url_str = str_from_list(urls)
+    logger.log(MESSAGE, "URLs `%s` removed from allow list.", url_str)
+    await ctx.respond(_("URLs `{}` removed from allow list.")
+                                .format(url_str))
+
```
Comment (jmercouris) [inline source/bouncer.py:119]: According to pep8 guidelines we should have two spaces between each top level function definition. I think it is time to add a linter! I'll see if I can figure out how to do that in the CI.

File: source/bouncer.py [hunk 4]
```diff
@@ -81,77 +86,58 @@ async def process_message(message):
...
-
-
-@client.event
-async def on_message_edit(message_before, message_after):
-    """Invoke when a message is edited on the Guild/server."""
-    if message_before.author.bot or message_after.author.bot:
-        return
-    await process_message(message_after)
+@client.slash_command()
+async def allow_domains(ctx, domains_separated_by_space: discord.Option(discord.SlashCommandOptionType.string)):
```
Comment (jmercouris) [inline source/bouncer.py:92]: ctx I assume to be the context? Maybe a docstring in the style of sphinx? https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html

**Comment by jmercouris** [comment]
I still have to package py-cord to test before merging!

File: source/bouncer.py (hunk 4 above)
Comment (nathan-jungels) [reply source/bouncer.py:92]: Yes. Agreed. I will add that.

File: requirements.txt (hunk 1 above)
Comment (nathan-jungels) [reply requirements.txt:4]: It was just moved here from your requirements file. Can it be removed?

Comment (jmercouris) [reply requirements.txt:4]: Yes, it is not needed.

**Comment by nathan-jungels** [comment]
Above comments have been acknowledged. Closing due to major changes.

**Pull request closed without merging**